src/lib.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::Result;
+use env::{create_env_virt, VirtEnv};
+use serde::Deserialize;
+use wit_component::ComponentEncoder;
+
+mod env;
+mod walrus_ops;
+
+#[derive(Deserialize, Debug, Default, Clone)]
+pub struct VirtOpts {
+    /// Environment virtualization
+    env: Option<VirtEnv>,
+}
+
+pub struct WasiVirt {
+    virt_opts: VirtOpts,
+}
+
+impl WasiVirt {
+    pub fn new() -> Self {
+        WasiVirt {
+            virt_opts: VirtOpts::default(),
+        }
```

Comment:
If `Default`, might as well be: 

```suggestion
        Self::default()
```